apple/resources.bzl

```diff
@@ -79,3 +84,81 @@ def apple_core_ml_library(name, mlmodel, **kwargs):
         data = [mlmodel],
         **kwargs
     )
+
+def objc_intent_library(
+    name,
+    src,
+    class_prefix = None,
+    class_visibility = None,
+    testonly = False,
+    swift_version = None,
+    **kwargs):
+    # buildifier: disable=function-docstring-args
+    """Macro to orchestrate an objc_library with generated sources for intentdefiniton files."""
+    intent_name = "{}.Intent".format(name)
+    intent_srcs = "{}.srcs".format(intent_name)
+    intent_hdrs = "{}.hdrs".format(intent_name)
+    _apple_intent_library(
+        name = intent_name,
+        src = src,
+        language = "Objective-C",
+        class_prefix = class_prefix,
+        header_name = name,
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    native.filegroup(
+        name = intent_srcs,
+        srcs = [intent_name],
+        output_group = "srcs",
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    native.filegroup(
+        name = intent_hdrs,
+        srcs = [intent_name],
+        output_group = "hdrs",
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    objc_library(
+        name = name,
+        srcs = [intent_srcs],
+        hdrs = [intent_hdrs],
+        sdk_frameworks = ["Intents"],
+        data = [src],
+        testonly = testonly,
+        **kwargs
+    )
+
+# Note: rules_apples depends on rules_swift, not the other way around. This means
+# that apple_intent_library could not be imported in rules_swift and thus this
+# macro must live here in rules_apple.
+def swift_intent_library(
+    name,
+    src,
+    class_prefix = None,
+    class_visibility = None,
+    swift_version = None,
+    testonly = False,
+    **kwargs):
+    # buildifier: disable=function-docstring-args
+    """Macro to orchestrate an swift_library with generated sources for intentdefiniton files."""
+    intent_name = "{}.Intent".format(name)
+    _apple_intent_library(
+        name = intent_name,
+        src = src,
+        language = "Swift",
+        class_prefix = class_prefix,
+        class_visibility = class_visibility,
+        swift_version = swift_version,
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    swift_library(
+        name = name,
```

Comment:
Can we expose an `module_name` and pass it down here? That way we can get something like `import SampleIntent` instead of `import app_SampleIntent`?

Reply:
kwargs are forwarded, so it is possible to pass a custom module_name or any swift_library attribute

Reply:
and if not specified, the standard module naming applies